### PR TITLE
Refactor signal documentation to improve clarity.

### LINKED ISSUE #97221: Signal is not created in specified object
### Tested versions

- reproducible in v4.3.stable.official [77dcf97d8]

### System information

Godot v4.3.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2070 SUPER (NVIDIA; 31.0.15.5186) - Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 Threads)

### Issue description

Documentation says:
```
Signal Signal(object: Object, signal: StringName)

Creates a new Signal named signal in the specified object.
```
But testing with this
```
extends Node

func _ready()->void:
	Signal(self, "damaged") # <----
	var signal_list:Array[Dictionary] = get_signal_list()
	for dict:Dictionary in signal_list:
		print(dict.name)
```
Results:
```
ready
renamed
tree_entered
tree_exiting
tree_exited
child_entered_tree
child_exiting_tree
child_order_changed
replacing_by
editor_description_changed
script_changed
property_list_changed
??? NO DAMAGED SIGNAL ???
```

I know that function returns a Signal, that seems to be bound to the object (using `print()`). But it's expected to be present in the signal list and accessed like a normal Signal.
If it's not part of objects signals, then what's the point to have an object reference as a parameter to `Signal(object, signal_name)`?
Signal() without provided parameters is useless, because it doesn't work when created in Array[Signal].
In result Signal construction is a bag of mixed signals.

### Steps to reproduce

- use function `Signal(self, "damaged")` to create a new signal for the object
- check if signal is created with `get_signal_list()`

### Minimal reproduction project (MRP)

[signal_not_created.zip](https://github.com/user-attachments/files/17071344/signal_not_created.zip)

### PR DESCRIPTION
**Pull request by Manik2607**
added the description mentioned in #97221.
please let me know if any changes are needed.

* *Bugsquad edit, fixes: #97221*

### CODE REVIEW

File: doc/classes/Signal.xml [hunk 1]
```diff
@@ -48,7 +48,7 @@
 			<param index="0" name="object" type="Object" />
 			<param index="1" name="signal" type="StringName" />
 			<description>
-				Creates a new [Signal] named [param signal] in the specified [param object].
+				Creates a signal object referencing a signal named [Signal] in the specified [param object].
```
Comment (AThousandShips) [inline doc/classes/Signal.xml:51]:
```suggestion
				Creates a [Signal] object referencing a signal named [param signal] in the specified [param object].
```

**Review by AThousandShips — APPROVED** [review]
Please remove the issue reference from the commit message

**Comment by AThousandShips** [comment]
Please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Comment by AThousandShips** [comment]
You broke your branch by a bad rebase, to fix this do:
```
git reset --hard 55262a6
```
And try again to rebase and squash

**Comment by AThousandShips** [comment]
You still need to squash your changes and push them with `git push --force`

**Comment by Manik2607** [comment]
i am really sorry for all the trouble  but i am really lost.
in my git log there is only 2 commits from me but on github it is saying 3 commits ahead.

55262a641e (HEAD -> master, origin/master, origin/HEAD) Refactor signal documentation to improve clarity
d5b37f76ec Merge branch 'godotengine:master' into master
4254946de9 (upstream/master) Merge pull request #97323 from timothyqiu/drop-unicode-identifier
2017006879 Merge pull request #97314 from tetrapod00/visual-shader-remap
2c1b8dda74 Merge pull request #97284 from timothyqiu/optimized-translation-gen
c47eb5c301 Merge pull request #97282 from timothyqiu/force-advanced-settings-when-searching

**Comment by AThousandShips** [comment]
There! Now everything is good!

**Comment by Manik2607** [comment]
Thank you for all the help. It would have taken you 2 minutes to fix but instead you helped me fix it.
I will try not to repeat these mistakes again.

**Comment by AThousandShips** [comment]
We try to use all opportunities for learning even if it takes a bit longer, good work figuring it out, git can be a maze

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**